### DISCUSSION TITLE
Shell commands in docker image are failing: Syntax error

### DISCUSSION THREAD

**Discussion by bosd** [joshreve/dactyl-keyboard#89] [Q&A] [answered]
When using the source form this repo, the command in the docker build is failling.

command:
`generate_configuration.py`
Generates a syntax error:
<details>
(myenv) root@50d8cf8d044b:/app/src# ./generate_configuration.py
./generate_configuration.py: line 1: import: command not found
./generate_configuration.py: line 2: import: command not found
./generate_configuration.py: line 3: import: command not found
./generate_configuration.py: line 4: import: command not found
./generate_configuration.py: line 7: pi: command not found
./generate_configuration.py: line 8: d2r: command not found
./generate_configuration.py: line 9: r2d: command not found
./generate_configuration.py: line 11: shape_config: command not found
./generate_configuration.py: line 13: ENGINE:: command not found
./generate_configuration.py: line 21: save_dir:: command not found
./generate_configuration.py: line 22: config_name:: command not found
./generate_configuration.py: line 24: show_caps:: command not found
./generate_configuration.py: line 25: show_pcbs:: command not found
./generate_configuration.py: line 27: nrows:: command not found
./generate_configuration.py: line 28: ncols:: command not found
./generate_configuration.py: line 30: alpha:: command not found
./generate_configuration.py: line 31: beta:: command not found
./generate_configuration.py: line 32: centercol:: command not found
./generate_configuration.py: line 33: centerrow_offset:: command not found
./generate_configuration.py: line 34: tenting_angle:: command not found
./generate_configuration.py: line 37: symmetry:: command not found
./generate_configuration.py: line 39: column_style_gt5:: command not found
./generate_configuration.py: line 40: column_style:: command not found
./generate_configuration.py: line 41: reduced_inner_cols:: command not found
./generate_configuration.py: line 42: reduced_outer_cols:: command not found
./generate_configuration.py: line 45: thumb_offsets:: command not found
./generate_configuration.py: line 46: syntax error near unexpected token `newline'
./generate_configuration.py: line 46: `    'keyboard_z_offset':  ('
</details>
`
generate_configuration.py ` also fails
<details>
(myenv) root@50d8cf8d044b:/app/src# ./generate_configuration.py
./generate_configuration.py: line 1: import: command not found
./generate_configuration.py: line 2: import: command not found
./generate_configuration.py: line 3: import: command not found
./generate_configuration.py: line 4: import: command not found
./generate_configuration.py: line 7: pi: command not found
./generate_configuration.py: line 8: d2r: command not found
./generate_configuration.py: line 9: r2d: command not found
./generate_configuration.py: line 11: shape_config: command not found
./generate_configuration.py: line 13: ENGINE:: command not found
./generate_configuration.py: line 21: save_dir:: command not found
./generate_configuration.py: line 22: config_name:: command not found
./generate_configuration.py: line 24: show_caps:: command not found
./generate_configuration.py: line 25: show_pcbs:: command not found
./generate_configuration.py: line 27: nrows:: command not found
./generate_configuration.py: line 28: ncols:: command not found
./generate_configuration.py: line 30: alpha:: command not found
./generate_configuration.py: line 31: beta:: command not found
./generate_configuration.py: line 32: centercol:: command not found
./generate_configuration.py: line 33: centerrow_offset:: command not found
./generate_configuration.py: line 34: tenting_angle:: command not found
./generate_configuration.py: line 37: symmetry:: command not found
./generate_configuration.py: line 39: column_style_gt5:: command not found
./generate_configuration.py: line 40: column_style:: command not found
./generate_configuration.py: line 41: reduced_inner_cols:: command not found
./generate_configuration.py: line 42: reduced_outer_cols:: command not found
./generate_configuration.py: line 45: thumb_offsets:: command not found
./generate_configuration.py: line 46: syntax error near unexpected token `newline'
./generate_configuration.py: line 46: `    'keyboard_z_offset':  ('
</details>

Environment:
Host os: Debian 10

**Comment by ivanmgribeiro** [comment, marked as answer]
What steps did you take to get to the point where this error is happening? I get similar errors when I launch the DM-shell container and try to run ```./generate_configuration.py``` in the shell that comes up. This is because docker is trying to execute the python file as a shell script.
I think you need to run ```python3 ./generate_configuration.py``` and ```python3 dactyl_manuform.py``` instead, and that should fix this issue.

You will probably also need to fix the issue mentioned in (https://github.com/joshreve/dactyl-keyboard/discussions/84), by changing the file as done here (https://github.com/ivanmgribeiro/dactyl-keyboard/commit/b71669bb069a68e890cf85229cfdbb2fc33bfaf8).

**Reply by bosd** [reply]
I've downloaded the files from your repo. Using the commands as outlined above and it works perfectly :smiley:
